modular/executor/migrate_task.go

```diff
@@ -219,21 +228,23 @@ func (e *ExecuteModular) doBucketMigrationReplicatePiece(ctx context.Context, gv
 	if err = e.baseApp.GfSpClient().ReplicatePieceToSecondary(ctx, destSPEndpoint, receive, data); err != nil {
 		log.CtxErrorw(ctx, "failed to replicate piece", "segment_piece_index", segmentIdx,
 			"redundancy_index", redundancyIdx, "error", err)
+		return err
 	}
 	return nil
 }
 
 func (e *ExecuteModular) doneBucketMigrationReplicatePiece(ctx context.Context, gvgID uint32, objectInfo *storagetypes.ObjectInfo,
-	params *storagetypes.Params, destSPEndpoint string, segmentIdx uint32) error {
+	params *storagetypes.Params, destSPEndpoint string, segmentIdx, redundancyIdx uint32) error {
 	receive := &gfsptask.GfSpReceivePieceTask{}
-	receive.InitReceivePieceTask(gvgID, objectInfo, params, coretask.DefaultSmallerPriority, segmentIdx, -1, 0)
+	receive.InitReceivePieceTask(gvgID, objectInfo, params, coretask.DefaultSmallerPriority, segmentIdx, int32(redundancyIdx), 0)
```

Comment:
receive.InitReceivePieceTask(gvgID, objectInfo, params, coretask.DefaultSmallerPriority,`0/* this arg useless*/`, int32(redundancyIdx), 0)